build/Dockerfile

```diff
@@ -1,43 +1,35 @@
-FROM alpine:3.14.6@sha256:06b5d462c92fc39303e6363c65e074559f8d6b1363250027ed5053557e3398c5
+FROM debian:buster-slim@sha256:5b0b1a9a54651bbe9d4d3ee96bbda2b2a1da3d2fa198ddebbced46dfdca7f216
+
+# Setting bash as our shell, and enabling pipefail option
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 # Some ENV variables
 ENV PATH="/mattermost/bin:${PATH}"
 ARG PUID=2000
 ARG PGID=2000
 ARG MM_PACKAGE="https://releases.mattermost.com/7.1.2/mattermost-7.1.2-linux-amd64.tar.gz?src=docker"
 
-
-# Install some needed packages
-RUN apk add --no-cache \
-  ca-certificates \
-  curl \
-  libc6-compat \
-  libffi-dev \
-  linux-headers \
-  mailcap \
-  netcat-openbsd \
-  xmlsec-dev \
-  tzdata \
-  wv \
-  poppler-utils \
-  tidyhtml \
-  && rm -rf /tmp/*
-
-# Get Mattermost
-RUN mkdir -p /mattermost/data /mattermost/plugins /mattermost/client/plugins \
```

Comment:
Thanks @phoinixgrr! Several tests failed due to unable to write file, during file upload. See https://mattermost-cypress-report.s3.amazonaws.com/228700-d7a7174-onprem-ent-server-pr-20807-mm-ee-test:d7a7174/mochawesome.html.

Reply:
Thanks for commenting @saturninoabril 
The good thing is that tests actually work now, the image building issue is now resolved 👍 


Concerning the failing tests,  
Would you mind letting me know where do we need to write a file? Maybe this is a permissions/dir not existing issue

Moreover, do you have a list of requirements that you could share, that the tests require from the image build? We can use those to make sure that the image build aligns 100%. 

Thanks

Reply:
As suspected the problem was permission related. 

Rebased, [force pushed changes](https://github.com/mattermost/mattermost-server/compare/d7a71748010dfbf4eed2bb38ef1794836758acb0..cd3a8ec4af403db512254e1c6662ebc70f32ffdb) and re-triggered e2e tests: 

https://mattermost-cypress-report.s3.amazonaws.com/228764-cd3a8ec-onprem-ent-server-pr-20807-mm-ee-test:cd3a8ec/mochawesome.html

Reply:
Thanks @phoinixgrr!

No special requirement for Cypress.

There were still failed tests related to image/video file upload. You may test it locally by trying a test user and upload `.tif` file. You may download https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/fixtures/tiff-image-file.tif and see if it works for you.

Reply:
@saturninoabril 
Any ideas how do we solve these failures? 
Would you know if these are related to https://community.mattermost.com/core/pl/88paijp4gjfpbykbrumej86noc ?

Reply:
Likely related to those removed packages but I'm not certain for sure. Maybe best to confirm from Server team.

Do you know why were those packages removed in the first place?

Reply:
@saturninoabril 
These packages were part of a 3 year old commit. 
 https://github.com/mattermost/mattermost-server/pull/11374/commits/fabfae0beb0b08319eacd257e2638ebbead6e1b7
 
Mattermost being a statically compiled go binary, it shouldn't have dependencies in system libs/bins.  
Moreover I checked our Mattermost-server install guide, I couldn't find any references dependencies/packages that should be installed for running it. 

Since this is also an effort, to optimize our container image, no need to install packages that are not needed .

Let me know your thoughts on this.

Reply:
@saturninoabril 

There are some references here: https://docs.mattermost.com/configure/file-storage-configuration-settings.html#enable-document-search-by-content but I am not sure if these are related, since are documented as optional dependencies. 

Is there a possibility that the tests fail without those?

Reply:
New test report after pushing: https://github.com/mattermost/mattermost-server/pull/20807/commits/cb3cb6629243611a7b10236135df4d8631fd7006

https://mattermost-cypress-report.s3.amazonaws.com/228845-cb3cb66-onprem-ent-server-pr-20807-mm-ee-test:cb3cb66/mochawesome.html

cc: @metanerd

Reply:
@saturninoabril  Any objections merging this?

Reply:
Thanks @phoinixgrr!